source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,293 @@
+   :language: java
+   :dedent:
+   :start-after: start findOneAndDelete-example
+   :end-before: end findOneAndDelete-example
+
+The output of the code above should look like this:
+
+.. code-block:: json
+
+   {"_id": 2, "food": "pear", "color": "yellow"}
+
+For more information on the ``Sorts`` class, see our
+:doc:`guide on the Sorts builder </fundamentals/builders/sort/>`.
+
+For more information on the methods and classes in this section, see the
+following API documentation pages:
+
+- :java-docs:`findOneAndDelete() <apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#findOneAndDelete(org.bson.conversions.Bson,com.mongodb.client.model.FindOneAndDeleteOptions)>`
+- :java-docs:`FindOneAndDeleteOptions <apidocs/mongodb-driver-core/com/mongodb/client/model/FindOneAndDeleteOptions.html>`
+
+Avoiding A Race Condition
+-------------------------
+
+In this section, we will go through an example that shows how we can
+avoid a race condition with compound operations.
+
+For this example, let's imagine that we run a hotel with one room and that we
+have a small Java program to help us checkout this room to a guest.  
+
+We represent our one room with this document in MongoDB. 
+
+.. code-block:: json
+
+    {"_id": 1, "guest": null, "room": "Blue Room", "reserved": false}
+
+`The full code for this example is available on Github here. <https://github.com/mongodb/docs-java/blob/master/source/includes/fundamentals/code-snippets/CompoundOperators.java>`__
+
+Example With Race Condition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's say our app uses this unsafe ``bookARoom`` method to checkout our room to
+a guest.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CompoundOperators.java
+   :language: java
+   :dedent:
+   :emphasize-lines: 3,11
+   :start-after: start the-unsafe-book-a-room
+   :end-before: end the-unsafe-book-a-room
+
+Imagine two separate guests, Jan and Pat, try to book the room with this method
+at the same time.
+
+Jan sees this output:
+
+.. code-block:: none
+
+   You got the Blue Room Jan
+
+And Pat sees this output:
+
+.. code-block:: none
+
+   You got the Blue Room Pat
+
+When we look at our database, we see the following: 
+
+.. code-block:: json
+
+    {"_id": 1, "guest": "Jan", "room": "Blue Room", "reserved": true}
+
+Pat will be unhappy. When Pat shows up to our hotel, Jan will be
+occupying her room. What went wrong?
+
+Here is the sequence of events that happened from the perspective of our MongoDB
+instance:
+
+- Find and return an empty room for Jan
+- Find and return an empty room for Pat
+- Update the room to booked for Pat
+- Update the room to booked for Jan
+
+Notice that for a brief moment Pat had reserved the room, but as Jan's update
+operation was the last to execute our document has ``"Jan"`` as the guest. 
+
+Let's see how we can use compound operations to avoid this race condition and
+always give our users the correct message. 
```

Comment:
Thanks for the note. This note was discussed synchronously over zoom.

We agreed that the section of text above was not performing enough work in the text, and that it should be reworded in order to replace first sentence of the avoiding a race condition section. We agreed that the sentence above should be rephrased to be less referential to the "Example With Race Condition" section so that readers could skip immediately to this section of the text.

Fix: